mksnapshot.sh

```diff
@@ -9,41 +9,43 @@
 
 echo "RUN: $0"
 if [ $# -lt 1 ]; then
-	echo "Usage: $0 [branch]"
-	echo "Where [branch] is the path under /bzr/ to the branch to snapshot."
+	echo "Usage: $0 [branch] [tag]"
+	echo "Where [branch] is the name of the branch to snapshot."
+	echo "Where [tag] is the name of the tag on that branch to snapshot."
 	exit 1
 fi
 # VCS details
-module=squid3
-BZRROOT=${BZRROOT:-/bzr}
-
-# generate a tarball name from the branch ($1) note that trunk is at
-# /bzr/trunk, but we call it 3.HEAD for consistency with CVS (squid 2.x), and
-# branches are in /bzr/branches/ but we don't want 'branches/' in the tarball
-# name so we strip that.
-branchpath=${1:-trunk}
-tag=${2:-`basename $branchpath`}
+if test "x${VCSROOT}" = "x"; then
+	echo "VCSROOT needs to be set to the base path for the repository to snapshot."
+	exit 1
+fi
+
+# generate a tarball name from the branch ($1) at tag ($2)
+branch=${1:-master}
+tag=${2:-HEAD}
 startdir=${PWD}
 date=`env TZ=GMT date +%Y%m%d`
 
-tmpdir=${TMPDIR:-${PWD}}/${module}-${tag}-mksnapshot
+tmpdir=${TMPDIR:-${PWD}}/squid-${tag}-mksnapshot/
 
 rm -rf ${tmpdir}
 trap "echo FAIL-BUILD_${VERSION} ; rm -rf ${tmpdir}" 0
+mkdir ${tmpdir}
 
 rm -f ${tag}.out
-bzr export ${tmpdir} ${BZRROOT}/${module}/${branchpath} || exit 1
+git checkout ${branch} || exit 1
+(git archive --format=tar ${tag} | tar -xC ${tmpdir}) || exit 1
 if [ ! -f ${tmpdir}/configure ] && [ -f ${tmpdir}/configure.ac ]; then
 	sh -c "cd ${tmpdir} && ./bootstrap.sh"
 fi
 if [ ! -f ${tmpdir}/configure ]; then
-	echo "ERROR! Tag ${tag} not found in ${module}"
+	echo "ERROR! Tag ${tag} not found."
 fi
 
 cd ${tmpdir}
-revision=`bzr revno ${BZRROOT}/${module}/${branchpath}`
+revision=`git rev-parse --short HEAD`
```

Comment:
1. Your item 1 can be a valid excuse but I am not convinced that using `git describe` would lead to (a lot) more changes or (significantly) increase complexity. The proposed `git rev-parse --short HEAD` gives your the latest commit SHA on the current branch while `git describe $branch` may be able to give you the right Squid version number (even for master!) and possibly the same SHA (if needed).
2. Your item 2 could be a side effect of us lacking the right tag for master. That tag would tell `git describe` that we are on a v5 path.
3. Your item 3 could be a side effect of your PR lacking a dedicated branch (normally, PRs originate from developer branches, not master). It could also be a side effect of how I fetched your PR -- I am still learning how to do that. In either case, it would not be applicable to branches.
4. I do not fully understand your item 4 but it seems to be irrelevant because `git describe` can be given a branch as a parameter.

Handling this properly requires understanding how `git describe` works and, possibly, adding a few tags. The long-term benefit may be significant, but this is not a blocking problem. If you do not want to study this any further, I certainly cannot insist that you do (because I do not know whether this investigation is likely to have a positive outcome).